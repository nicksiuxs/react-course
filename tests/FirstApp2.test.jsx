import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FisrtApp";

describe("Test in FirstApp.test.jsx", () => {
  const title = "Hola, soy un titulo";
  const subtitle = "Soy un subtitulo";

  test("Should match the component with the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Should show the 'Hola, soy un titulo ", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Should show the title in a h1 tag", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Should show the subtitle sent in props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getByText(subtitle)).toBeTruthy();
  });
});
