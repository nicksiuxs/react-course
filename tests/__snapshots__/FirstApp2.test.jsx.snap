// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp.test.jsx Should match the component with the snapshot 1`] = `
<div>
  <h1
    data-testid="title"
  >
    Hola, soy un titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay nombre
  </p>
</div>
`;
